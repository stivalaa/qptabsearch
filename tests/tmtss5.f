*=======================================================================
* File:    tmtest5.f
* Author:  Alex Stivala
* Created: July 2008
*
* Test program for tmatn.f
*
* $Id: tmtss5.f 2971 2009-11-22 01:50:50Z astivala $
*=======================================================================

      program tmtest5

      implicit none
      
      external tmatn

      integer n1,n2,info
      parameter(n1 = 7, n2 = 13)
      double precision omega1(n1,n1),omega2(n2,n2)
      double precision dmat1(n1,n1),dmat2(n2,n2)
      double precision score
      double precision soln(n1*n2+n1+n2)

*     MATLAB GetTableau('1ABA')
      data omega1  / 0 ,  -2.8600 ,  -0.3600 ,  -2.2100 ,
     $     2.9200 ,  -0.8100 ,  -2.7700,
     $     -2.8600 ,   1.0000 ,  -2.9100 ,   1.1200 ,
     $     -0.4500 ,   2.1300 ,   0.3300,
     $     -0.3600 ,  -2.9100 ,        0 ,  -2.1800 ,   2.5600 ,
     $     -1.1600 ,  -0.1200,
     $     -2.2100 ,   1.1200 ,  -2.1800 ,
     $     1.0000 ,   1.0200 ,  -1.8800 ,   0.8400,
     $     2.9200 ,  -0.4500 ,   2.5600 , 
     $     1.0200 ,        0 ,   2.5600 ,   0.6000,
     $     -0.8100 ,   2.1300 ,  -1.1600 ,
     $     -1.8800 ,   2.5600 ,        0 ,   1.9700,
     $     -2.7700 ,   0.3300 ,  -0.1200 ,
     $     0.8400 ,   0.6000 ,   1.9700 ,   1.0000 /


* pytableaucreate.py -t dssp -n /var/tmp/d1xdtt3.ent
      data omega2 /0.00, -1.11,  1.78, -1.74, -1.65, -2.49,
     $     1.46,  1.89,  1.53, -1.24, -0.84,  2.70,  2.40, 
     $     -1.11,  0.00,  2.89,  1.16, -1.02, -2.59,  1.39,
     $     -0.15,  1.27, -1.70, -0.39,  2.42,  1.96, 
     $     1.78,  2.89,  0.00, -2.06,  2.21, -0.79, -1.72,
     $     0.11, -1.88 , 1.35,  2.55, -0.96, -1.35, 
     $     -1.74,  1.16, -2.06,  0.00,  1.88,  1.95, -0.51,
     $     2.06,  0.32, -2.86, -1.52 , 1.47,  0.86, 
     $     -1.65, -1.02 , 2.21 , 1.88 , 0.00 , 1.66,
     $     2.33 ,-2.15 , 2.15 ,-1.06 ,-0.95 , 1.89 , 2.10 ,
     $     -2.49 ,-2.59 ,-0.79 , 1.95 , 1.66 , 0.00 ,-1.99,
     $     -0.68 , 2.03 ,-1.43 ,-2.60 , 0.48 , 1.10 ,
     $     1.46 , 1.39 ,-1.72 ,-0.51 , 2.33 ,-1.99 , 0.00,
     $     -1.76 ,-0.19 , 2.67 , 1.65 ,-1.54 ,-0.96 ,
     $     1.89 ,-0.15 , 0.11 , 2.06 ,-2.15 ,-0.68 ,-1.76,
     $     0.00 ,-1.91 , 1.36 , 2.62 ,-0.87 ,-1.30 ,
     $     1.53 , 1.27 ,-1.88 , 0.32 , 2.15 , 2.03 ,-0.19,
     $     -1.91 , 0.00 , 2.78 , 1.56 , 1.55 , 0.94 ,
     $     -1.24 ,-1.70 , 1.35 ,-2.86 ,-1.06 ,-1.43 , 2.67,
     $     1.36 , 2.78 , 0.00 ,-1.34 ,-1.91 ,-2.53 ,
     $     -0.84 ,-0.39 , 2.55 ,-1.52 ,-0.95 ,-2.60 , 1.65,
     $     2.62 , 1.56 ,-1.34 , 0.00 , 2.74 , 2.35 ,
     $     2.70 , 2.42 ,-0.96 , 1.47 , 1.89 , 0.48 ,-1.54 ,-0.87,
     $     1.55 ,-1.91 , 2.74 , 0.00 ,-0.62 ,
     $     2.40 , 1.96 ,-1.35 , 0.86 , 2.10 , 1.10 ,-0.96,
     $     -1.30 , 0.94 ,-2.53 , 2.35 ,-0.62 , 0.00 /


      data dmat1 / 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0 /


      data dmat2 / 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 /

*     should get score approx -5.5978 (MATLAB TAbleauMAtch)

      call tmatn(omega1, n1, n1, omega2, n2, n2, .false., .false.,
     $           dmat1, dmat2,
     $           score, soln, info)
      if (info .ne. 0) then
         write(*,*),'TABMATN FAILED'
      else
         write(*,900),score
 900     format('score = ',f6.2)
      endif
      end

