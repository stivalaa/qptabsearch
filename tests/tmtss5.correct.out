score =  -6.62
